### PR TITLE
fix(cni): hook up log level to cni

### PR DESCRIPTION
**Pull request by slonka**
Signed-off-by: slonka <slonka@users.noreply.github.com>

The old CNI and the new experimental CNI did not have adjustable log level, this PR fixes it for the new CNI.

### Checklist prior to review

- [X] Link to docs PR or issue -- PR in progress https://github.com/kumahq/kuma-website/pull/958
- [X] Link to UI issue or PR -- not a UI issue
- [X] Is the [issue worked on linked][1]? -- not reported yet
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [X] E2E Tests --
- [X] Manual Universal Tests --
- [X] Manual Kubernetes Tests -- manually tested this on k3d
- [X] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? -- no
- [X] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? -- no

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/cni/pkg/install/main.go [hunk 1]
```diff
@@ -25,17 +26,26 @@ const (
...
 func removeBinFiles() error {
 	return os.Remove("/host/opt/cni/bin/kuma-cni")
 }
 
 func cleanup(ic *InstallerConfig) {
+	cleanupMutex.Lock()
+	defer cleanupMutex.Unlock()
+	if cleanupFinished == true {
+		return
+	}
```
Comment (bartsmykla) [inline app/cni/pkg/install/main.go:47]:
```suggestion
	if cleanupFinished {
		return
	}
```

File: app/cni/pkg/install/utils.go [hunk 2]
```diff
@@ -22,3 +28,17 @@ func parseBytesToHashMap(bytes []byte) (map[string]interface{}, error) {
...
+
+func SetLogLevel(logger *logr.Logger, level string, name string) {
+	logLevel, err := kuma_log.ParseLogLevel(level)
+
+	if err != nil {
+		logLevel = kuma_log.InfoLevel
+	}
+
+	*logger = CreateNewLogger(name, logLevel)
+}
```
Comment (bartsmykla) [inline app/cni/pkg/install/utils.go:44]:
```suggestion
}

```

**Review by bartsmykla — APPROVED** [review]
few nits, but overall lgtm

File: app/cni/pkg/install/main.go [hunk 3]
```diff
@@ -25,17 +26,26 @@ const (
...
 func removeBinFiles() error {
 	return os.Remove("/host/opt/cni/bin/kuma-cni")
 }
 
 func cleanup(ic *InstallerConfig) {
+	cleanupMutex.Lock()
+	defer cleanupMutex.Unlock()
+	if cleanupFinished {
+		return
+	}
```
Comment (jakubdyszkiewicz) [inline app/cni/pkg/install/main.go:47]: Does that mean that cleanup is called multiple times?

File: app/cni/pkg/install/utils.go [hunk 4]
```diff
@@ -22,3 +28,17 @@ func parseBytesToHashMap(bytes []byte) (map[string]interface{}, error) {
...
 }
+
+func CreateNewLogger(name string, logLevel kuma_log.LogLevel) logr.Logger {
+	return core.NewLoggerTo(os.Stderr, logLevel).WithName(name)
+}
+
+func SetLogLevel(logger *logr.Logger, level string, name string) {
+	logLevel, err := kuma_log.ParseLogLevel(level)
+
+	if err != nil {
```
Comment (jakubdyszkiewicz) [inline app/cni/pkg/install/utils.go:39]: why not return an error?

Comment (slonka) [reply app/cni/pkg/install/utils.go:39]: If we want to quit if invalid level is set then yes we can return an error.

File: app/cni/pkg/install/main.go (hunk 3 above)
Comment (slonka) [reply app/cni/pkg/install/main.go:47]: we do a `defer` and "signal exit" here - https://github.com/kumahq/kuma/blob/ad73606e83780877ace03c0b3cddd2cb0029caa4/app/cni/pkg/install/main.go#L256 - I'm not sure `defer` will always be called if a process is killed via signal, so I saw in the logs that it was called multiple times. If there are guarantees on `defer` being called I can remove the signal cleanup.

File: app/cni/pkg/install/utils.go [hunk 5]
```diff
@@ -22,3 +28,17 @@ func parseBytesToHashMap(bytes []byte) (map[string]interface{}, error) {
 	}
 	return parsed, nil
 }
+
+func CreateNewLogger(name string, logLevel kuma_log.LogLevel) logr.Logger {
+	return core.NewLoggerTo(os.Stderr, logLevel).WithName(name)
```
[the remark this answers is not in the document]
Comment (slonka) [reply app/cni/pkg/install/utils.go:33]: It was explained in the #4845 but I will drop a comment in the code as well.

File: app/cni/pkg/install/utils.go (hunk 4 above)
Comment (slonka) [reply app/cni/pkg/install/utils.go:39]: If we want to quit the app if the wrong log level is set then we could do that.

File: app/cni/pkg/install/utils.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply app/cni/pkg/install/utils.go:33]: I don't think it does. It's just unusual to produce all the logs on stderr by default. I remember the reason but someone who will read the code in the future may not.

File: app/cni/pkg/install/utils.go (hunk 4 above)
Comment (jakubdyszkiewicz) [reply app/cni/pkg/install/utils.go:39]: We do this in Kuma CP / other binaries and I think it makes sense. It protects the user from typos

**Review by jakubdyszkiewicz — APPROVED** [review]

**Comment by slonka** [comment]
@Mergifyio backport release-1.8

**Pull request merged**